src/databricks/labs/dqx/check_funcs.py

```diff
@@ -2004,23 +2107,100 @@ def _convert_ipv4_to_bits(ip_col: Column) -> Column:
     return _extract_octets_to_bits(ip_col, DQPattern.IPV4_ADDRESS.value)
 
 
-def _convert_cidr_to_bits_and_prefix(cidr_col: Column) -> tuple[Column, Column]:
+def _convert_ipv4_cidr_to_bits_and_prefix(cidr_col: Column) -> tuple[Column, Column]:
     """Returns binary IP and prefix length from CIDR (e.g., '192.168.1.0/24')."""
     ip_bits = _extract_octets_to_bits(cidr_col, DQPattern.IPV4_CIDR_BLOCK.value)
     # The 5th capture group in the regex pattern corresponds to the CIDR prefix length.
     prefix_length = F.regexp_extract(cidr_col, DQPattern.IPV4_CIDR_BLOCK.value, 5).cast("int").alias("prefix_length")
     return ip_bits, prefix_length
 
 
-def _get_network_address(ip_bits: Column, prefix_length: Column) -> Column:
+def _get_network_address(ip_bits: Column, prefix_length: Column, total_bits: int) -> Column:
     """
     Returns the network address from IP bits using the CIDR prefix length.
 
     Args:
-        ip_bits: 32-bit binary string representation of the IPv4 address
+        ip_bits: Binary string representation of the IP address (32 or 128 bits)
         prefix_length: Prefix length for CIDR notation
+        total_bits: Total number of bits in the IP address (32 for IPv4)
+
+    Returns:
+        Network address as a binary string of the same total length
+    """
+    return F.rpad(F.substring(ip_bits, 1, prefix_length), total_bits, "0")
+
+
+def _build_is_valid_ipv6_address_udf() -> Callable:
+    """
+    Build a user-defined function (UDF) to check if a string is a valid IPv6 address.
 
     Returns:
-        Network address as a 32-bit binary string
+        Callable: A UDF that checks if a string is a valid IPv6 address
     """
-    return F.rpad(F.substring(ip_bits, 1, prefix_length), 32, "0")
+
+    @F.pandas_udf("boolean")  # type: ignore[call-overload]
+    def _is_valid_ipv6_address_udf(column: pd.Series) -> pd.Series:
+        # Self-contained validation logic to avoid serialization issues
+        def is_valid_ipv6_local(ip_str):
+            if pd.isna(ip_str) or ip_str is None:
+                return False
+            import ipaddress  # pylint: disable=import-outside-toplevel
```

Comment:
Had to add `disable=import-outside-toplevel` if not `make lint` complains about the import not being on outside level